### PR TITLE
Add treemacs layer.

### PR DESCRIPTION
**Pull request by Alexander-Miller**
A layer for [treemacs](https://github.com/Alexander-Miller/treemacs), as discussed [here](https://github.com/syl20bnr/spacemacs/issues/9285).

### CODE REVIEW

**Comment by fiveNinePlusR** [comment]
I would say this is a big win for spacemacs to get it merged into develop. Great package and nice integration. pretty low friction replacement for neotree.

**Comment by JAremko** [comment]
Would be nice to have screenshot in the documentation (not too big one) :thinking:

**Comment by syl20bnr** [comment]
And now we have the shadow mechanism to easily mark a layer to replace another.
I'll have time to merge some PRs at the end of the week, this PR will be part of the batch. I'll merge it myself as I have some reorganization to do with neotree as well.

Good job!

**Comment by Alexander-Miller** [comment]
> And now we have the shadow mechanism to easily mark a layer to replace another.

Anything that I need to do here?

> Would be nice to have screenshot in the documentation

Added one to the description.

File: layers/+tools/treemacs/README.org [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+#+BEGIN_SRC emacs-lisp
+  (treemacs
+   :variables
+   treemacs-use-follow-mode t
+   treemacs-use-filewatch-mode t
+   treemacs-use-collapsed-directories 3)
+#+END_SRC
+
+* Configuraion
```
Comment (nightscape) [inline layers/+tools/treemacs/README.org]: Typo

Comment (Alexander-Miller) [reply layers/+tools/treemacs/README.org]: Fixed it now.

**Comment by syl20bnr** [comment]
Thank you ! 👍 
I moved the layer to `+filetree` directory and made some formatting fixes in README to make it more idiomatic.
I had to add `treemacs--window-number-ten` explicitly in `use-package` treemacs init otherwise I had breaking errors when helm is invoked before emacstree.

Nice job 💙 

Cherry-picked into develop branch, you can safely delete your branch.

**Comment by Alexander-Miller** [comment]
Thanks for the help.

I looked around the new files and it looks like neotree's readme is listing features from the python layer. The linked image isnt there either. I'd put up a PR, but I dont know whats supposed to be in that list.

**Comment by syl20bnr** [comment]
Thank you for noting this :-)
I can fix the list myself.

**Comment by syl20bnr** [comment]
@Alexander-Miller I pushed a fix, thanks again !

**Comment by duianto** [comment]
Does Treemacs in Spacemacs require python?
It's working fine with `emacs --quick`, `M-x package-initialize` and `M-x treemacs`.

In Spacemacs, this message is shown, the first time `SPC f t` is pressed, and every time `SPC f T` is pressed:
`apply: Searching for program: No such file or directory, python`

The treemacs window only shows the `*Roaming*` parent directory, when treemacs was called from `.spacemacs`, no directories or files are shown below the `*Roaming*` line.

These might be related to the python error:
- Trying to navigate with either the `vim` or `emacs` editing styles, doesn't work, emacs style reports:
`funcall-interactively: Buffer is read-only: #<buffer *Treemacs*>`
and with the vim style, the cursor moves left and right on the `*Roaming*` line, and up and down reports:
evil-line-move: Beginning of buffer
evil-line-move: End of buffer

- With the `vim` style, pressing `?` opens the `evil-ex-search-backward` "?" minibuffer prompt, instead of the hydra. With the `emacs` style, `?` shows:
`funcall-interactively: Buffer is read-only: #<buffer *Treemacs*>`

The following was only been tested in `emacs --quick`:
The hydra closes as soon as a key is pressed, ex: `n` to move to the next line. If it would stay open until it was closed manually by pressing `?` again, then one could try the key bindings while the hydra is open, without having to open it between each key press. That would also match the behaviour of the other spacemacs hydras.

#### System Info :computer:
- OS: windows-nt
- Emacs: 26.0.50
- Spacemacs: 0.200.9
- Spacemacs branch: develop (rev. 07014dee)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-enable-help-tooltip 'manual auto-completion-enable-snippets-in-popup t)
 autohotkey emacs-lisp git helm markdown org spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES

**Comment by Alexander-Miller** [comment]
> Does Treemacs in Spacemacs require python?

The collapse directory feature does. Set the collapse-dirs value to 0 to disable it. I've already an update for the readme in the works, Ill add the part about python and adjust the default value accordingly.

Ill look at the other issues when I get home.

**Comment by duianto** [comment]
That fixed it. 👍 
Navigation works, and `?` toggles the hydra, with both the emacs and vim editing style.

Maybe one solution could be to set it to `0`, if emacs is unable to find python.

**Comment by Alexander-Miller** [comment]
> Maybe one solution could be to set it to 0, if emacs is unable to find python.

Opened a PR to do just that. Ill also put a more permanent hydra on my todo list.

**Comment by robbert-vdh** [comment]
It would also be nice if the treemacs layer bound `SPC p t` instead of `SPC f p/P` to `treemacs-projectile-toggle`  so it's consistent with the existing neotree keybindings.

**Comment by Alexander-Miller** [comment]
I tried just that at first, but it didnt work. pt was always bound to `projectile-find-tag`. Thats why I went with fp instead.

**Pull request closed without merging**